### PR TITLE
[WIP] New install commands

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Fixed tickets | #1734 and few more probably
| License       | MIT
| Doc PR        | Pushed to master

install

See the docs http://docs.sylius.org/en/latest/cookbook/installation-commands.html

### CODE REVIEW

File: app/console [hunk 1]
```diff
@@ -13,7 +13,7 @@
 set_time_limit(0);
 
 require_once __DIR__.'/bootstrap.php.cache';
-require_once __DIR__.'/CliKernel.php';
+require_once __DIR__.'/AppKernel.php';
```
Comment (stloyd) [inline app/console]: Why?

File: src/Sylius/Bundle/InstallerBundle/Command/CheckRequirementsCommand.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Sylius\Bundle\InstallerBundle\Command;
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/CheckRequirementsCommand.php]: Missing license.

File: src/Sylius/Bundle/InstallerBundle/Command/CheckRequirementsCommand.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+use RuntimeException;
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/CheckRequirementsCommand.php]: In all place we use `\RuntimeException`...

File: src/Sylius/Bundle/InstallerBundle/Command/CommandExecutor.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+     */
+    public function __construct(InputInterface $input, OutputInterface $output, Application $application)
+    {
+        $this->input = $input;
+        $this->output = $output;
+        $this->application = $application;
+    }
+
+    /**
+     * @param $command
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/CommandExecutor.php]: Missing type, also one param is missing too.

File: src/Sylius/Bundle/InstallerBundle/Command/CommandExecutor.php [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            $this->getDefaultParameters()
+        );
+
+        $this->application->setAutoExit(false);
+        $exitCode = $this->application->run(new ArrayInput($parameters), $output ?: new NullOutput());
+
+        if (0 !== $exitCode) {
+            $this->application->setAutoExit(true);
+
+            $errorMessage = sprintf('The command terminated with an error code: %u.', $exitCode);
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/CommandExecutor.php]: Why not putting `<error>` here already?

File: src/Sylius/Bundle/InstallerBundle/Command/CommandExecutor.php [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            $this->output->writeln("<error>$errorMessage</error>");
+            $exception = new \Exception($errorMessage, $exitCode);
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/CommandExecutor.php]: Why variable as it's one time use?

File: src/Sylius/Bundle/InstallerBundle/Command/InstallAssetsCommand.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Sylius\Bundle\InstallerBundle\Command;
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/InstallAssetsCommand.php]: Missing license.

File: app/console (hunk 1 above)
Comment (pjedrzejewski) [inline app/console]: I think the benefit was minor, it was confusing compared to Symfony standard and generally running commands with a different set of bundles than web instance sounds fragile to me. I was also often getting issue when generating migrations. I think we are fine with these 3 bundles in default kernel. Makes things simpler.

File: src/Sylius/Bundle/InstallerBundle/Command/AbstractInstallCommand.php [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+            }
+        }
+
+        if ($displayProgress) {
+            $progress->finish();
+        }
+    }
+
+    /**
+     * @param string $question
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/AbstractInstallCommand.php]: Not matches...

File: src/Sylius/Bundle/InstallerBundle/Command/AbstractInstallCommand.php [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+     *
+     * @return boolean
+     */
+    protected function validate($value, array $constraints = array())
+    {
+        return $this->get('validator')->validateValue($value, $constraints);
+    }
+
+    /**
+     * @param ConstraintViolationList $errors
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/AbstractInstallCommand.php]: Not matches.

File: src/Sylius/Bundle/InstallerBundle/Command/InstallCommand.php [hunk 10]
```diff
@@ -1,139 +1,69 @@
...
 
-    protected function runCommand($command, InputInterface $input, OutputInterface $output)
-    {
-        $this
-            ->getApplication()
-            ->find($command)
-            ->run($input, $output)
-        ;
+        $map = array(
+            Kernel::ENV_DEV => '/app_dev.php',
```
Comment (stloyd) [inline src/Sylius/Bundle/InstallerBundle/Command/InstallCommand.php]: Really? Adding dependency to get just __4__ hardcoded variables? IMO better would be to use this as configuration or even in `parameters.yml`...

Comment (pjedrzejewski) [inline src/Sylius/Bundle/InstallerBundle/Command/InstallCommand.php]: Installer is only for Core, so I'm not sure what is the problem here. The dependency was missing. We do not want a generic installer, cause it is not really easy to do and maintain.

**Comment by psyray** [comment]
Hi,

New issue due to this PR merge
Doctrine migration broken : #2465 

Thanks to fix asap

**Pull request merged**